servers/republik/modules/crowdfundings/graphql/resolvers/User.js

```diff
@@ -75,6 +77,73 @@ module.exports = {
     }
     return false
   },
+  async customPackages (user, args, { pgdb, user: me }) {
+    debug('customPackages')
+
+    Roles.ensureUserIsMeOrInRoles(user, me, ['supporter'])
+
+    const now = new Date()
+
+    const crowdfunding = args.crowdfundingName
+      ? await pgdb.public.crowdfundings.findOne({ name: args.crowdfundingName })
+      : await pgdb.public.crowdfundings.findOne({
+        'beginDate <=': now,
+        'endDate >': now
+      })
+
+    const packages = await pgdb.public.packages.find({
+      crowdfundingId: crowdfunding.id,
+      custom: true
+    })
+
+    const pledges = await pgdb.public.pledges.find({
+      userId: user.id,
+      status: 'SUCCESSFUL'
+    })
+
+    const memberships =
+      pledges.length > 0
+        ? await pgdb.public.memberships.find({
+          pledgeId: pledges.map(pledge => pledge.id)
+        })
+        : []
+
+    const membershipTypes =
+      memberships.length > 0
+        ? await pgdb.public.membershipTypes.find({
+          id: memberships.map(membership => membership.membershipTypeId)
+        })
+        : []
+
+    memberships.forEach((membership, index, memberships) => {
+      memberships[index].membershipType =
+        membershipTypes.find(membershipType => membershipType.id === membership.membershipTypeId)
+    })
+
+    const membershipPeriods =
+      memberships.length > 0
+        ? await pgdb.public.membershipPeriods.find({
+          membershipId: memberships.map(membership => membership.id)
+        })
+        : []
+
+    memberships.forEach((membership, index, memberships) => {
+      memberships[index].membershipPeriods =
+        membershipPeriods.filter(membershipPeriod => membershipPeriod.membershipId === membership.id)
+    })
+
+    Object.assign(user, { memberships })
```

Comment:
why do you mutate user here? you are in a recursive resolver flow, downstream resolvers might get confused.